### PR TITLE
Write docs on AuthSigs, SessionSigs, PKPs

### PR DESCRIPTION
**Pull request by hwrdtm**
# What

Related to LIT-168.
Closes LIT-140.
Closes LIT-514.

Also did some sidebar restructuring.

# TODO

- There are some things I've left marked as `TODO:` because I've communicated with @sarahzdev that it's better that she help fill in when her `lit-auth-client` package is ready.
- I think the `pkp/authHelpers` page needs a revisit later on.
- I think the `SDK/Explanation/authentication/sessionSigs` page needs a revisit later on when we fix up + unify sessionSigs across our entire SDK. 
- [x] Reconcile with this section https://github.com/LIT-Protocol/docs/pull/76/files#diff-1dce7e3b27d8e9edaf05442cb2ac4281d69ae494395e6822ff5ca623efc6a19fR22
  - [x] Pull out session key to session sigs intro
  - [x] Can have brief security comments inline in `/getSessionSigs` page but call out to Security Considerations page for more details.
  - [x] Go in detail about the types of capabilities supported (in security section?)
  - [x] Omit `pkpPublicKey`.

### CODE REVIEW

File: docs/SDK/Explanation/authentication/methods.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Authentication Methods
+
+There are various authentication methods to obtain an AuthSig:
```
Comment (sarahzdev) [inline docs/SDK/Explanation/authentication/methods.md:3]: wondering if we could better differentiate between an authsig for a pkp vs. an authsig used to obtain session sigs for a pkp.

at the moment, it seems like we're warning them against using authsigs and then saying to generate authsigs for session sigs.

File: docs/SDK/Explanation/authentication/sessionSigs.md [hunk 2]
```diff
@@ -2,25 +2,132 @@
...
 
+The session keypair is used to sign all requests to the Lit Protocol API, and the user's AuthSig is sent along with the request, attached as a "capability" to the session signature. Each node in the Lit Network receives a unique signature for each request, and can verify that the user owns the wallet address that signed the capability.
+
+### An Improved User Experience
+
+The steps to obtain an AuthSig requires an interactive experience that involves manual steps from the end-user - whether clicking through the Google OAuth flow, placing their fingerprint on their platform authenticators, or clicking through their the respective modals of their externally-owned accounts (eg. MetaMask). These manual steps can be friction that drives users away from applications.
+
+On the other hand, the steps to obtain a SessionSig is completely non-interactive. The session keys and signature can all be done programmatically once an AuthSig has been obtained.
```
Comment (sarahzdev) [inline docs/SDK/Explanation/authentication/sessionSigs.md:29]: the "On the other hand" comparison seems to suggest that the end user may not be involved in obtaining a session sig, but the manual steps are still required to get the args for generating the session sigs.

and the "Format of SessionSigs" seems to imply that sessionSigs are really just signed authSigs.

File: docs/pkp/minting.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+## Technical Details
+
+### Contract Specifics
+
+- The `authMethodId` is derived from the credential's [rawId](https://www.w3.org/TR/webauthn-2/#dom-publickeycredential-rawid) parameter.
+- The `authMethodPubkey` is the COSE credential public key.
+
+### Relying Parties and Supported Origins
+
+In order to allow for various frontends to integrate with our platform, we plan to support any domain to act as a [Relying Party](https://www.w3.org/TR/webauthn-2/#webauthn-relying-party) in the long run. However, we are in the process of slowly rolling out this authentication method currently maintain an allowlist of origins / domains that can integrate with the Lit network.
```
Comment (joshLong145) [inline docs/pkp/minting.md:32]: Should we mention that nodes will require `localhost:3000` to be the `rpId`

File: docs/pkp/minting.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+We have a frontend that helps with this process at https://pkp-walletconnect.vercel.app/.
+
+We currently support both username-based and username-less WebAuthn registration, and usernames are purely used for your convenience / reference on the client-side.
+
+## Technical Details
+
+### Contract Specifics
+
+- The `authMethodId` is derived from the credential's [rawId](https://www.w3.org/TR/webauthn-2/#dom-publickeycredential-rawid) parameter.
+- The `authMethodPubkey` is the COSE credential public key.
```
Comment (joshLong145) [inline docs/pkp/minting.md:28]: should there be something about how our nodes can only support validating with ECDSA keys?

**Review by sarahzdev — APPROVED** [review]

File: docs/SDK/Explanation/authentication/methods.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+## WebAuthn Authentication
+
+The steps below demonstrate how to authenticate using WebAuthn in order to obtain an AuthSig.
+
+TODO:
+
+### Technical Details
+
+While the registration step involves minting a PKP via the Lit Relay server (or, optionally, you can send a transaction to the contract yourself), we have implemented an authentication scheme that involves sending signed challenges to the decentralized Lit network instead. This works by using a recent blockHash on the underlying blockchain (Polygon Mumbai) as a challenge, and having the user authenticate with their platform authenticator to generate a credential assertion (signature). When each Lit node receives this credential assertion from the client, they can recover the COSE credential public key which is stored in the smart contract to verify whether the assertion / signature is valid. If the signature is valid, then the nodes will return AuthSig signature shares back to the client.
```
Comment (debbly) [inline docs/SDK/Explanation/authentication/methods.md:32]: I think we talk about challenges in the PKP section as well (not covered in this PR), and we need a definition around what a challenge is.

File: docs/SDK/Explanation/authentication/authSig.md
Comment (debbly) [inline docs/SDK/Explanation/authentication/authSig.md:1]: the netlify.uml file should be updated to redirect the old authSig link to the new one

old `/SDK/Explanation/WalletSigs/authSig` 

new link `/SDK/Explanation/authentication/authSig`

File: docs/SDK/Explanation/authentication/authSig.md [hunk 6]
```diff
@@ -6,15 +6,19 @@ sidebar_position: 1
 
 ## AuthSigs
 
-To use Lit Protocol, you must present a wallet signature obtained from the user. This is refered to an as `AuthSig` in the documentation. You can use any EIP 4361 compliant signature (Sign in with Ethereum) for the authSig, but you must put the signature into the AuthSig data structure format (documented [here](https://js-sdk.litprotocol.com/interfaces/types_src.JsonAuthSig.html)). You do not need to use the Lit JS SDK V2 to obtain the signature as long as it's EIP 4361 compliant and in the AuthSig data structure format.
+We refer to a wallet signature obtained from the user as an AuthSig.
+
+You can use any EIP 4361 compliant signature (Sign in with Ethereum) for the authSig, but you must put the signature into the AuthSig data structure format (documented [here](https://js-sdk.litprotocol.com/interfaces/types_src.AuthSig.html)). You do not need to use the Lit JS SDK V2 to obtain the signature as long as it's EIP 4361 compliant and in the AuthSig data structure format.
```
Comment (debbly) [inline docs/SDK/Explanation/authentication/authSig.md:11]: link reference to EIP 4361 at first mention of it - https://eips.ethereum.org/EIPS/eip-4361

File: docs/SDK/Explanation/authentication/sessionSigs.md
Comment (debbly) [inline docs/SDK/Explanation/authentication/sessionSigs.md:1]: Same comment as above with adding the to netlify.uml for the redirect

old link: `/SDK/Explanation/WalletSigs/sessionSigs`

new link: `/SDK/Explanation/authentication/sessionSigs`

File: docs/SDK/Explanation/authentication/sessionSigs.md [hunk 7]
```diff
@@ -136,23 +249,50 @@ const decryptedString = await decryptedFiles["string.txt"].async(
...
 
 The `getSessionSigs()` function will generate a session key for you automatically and attempt to store it in LocalStorage. In case you have not polyfilled LocalStorage, you may instead generate the session key yourself using `generateSessionKeyPair()` and store it however you like. You can then pass it to `getSessionSigs()` as the `sessionKey` param.
 
-## Obtaining session signatures when the user doesn't have a wallet
+## Obtaining the SessionSig when user doesn't have a wallet
 
-You can use Oauth login with services including Discord and Google when the user doesn't have a wallet. You can see an example of how to do this using Google Oauth here: https://github.com/LIT-Protocol/oauth-pkp-signup-example/blob/main/src/App.tsx#L386
+You can use Oauth login with services including Discord and Google when the user doesn't have a wallet. You can see an example of how to do this using Google Oauth here: https://github.com/LIT-Protocol/oauth-pkp-signup-example/blob/main/src/App.tsx#L50
```
Comment (debbly) [inline docs/SDK/Explanation/authentication/sessionSigs.md:260]: let's update this to have the link with [here](), vs having the full link in the doc

File: docs/SDK/Explanation/tests.md [hunk 8]
```diff
@@ -53,10 +53,10 @@ Errors are thrown as exceptions when something has gone wrong. Errors are object
 |NodeExpWrongOrTooLarge|The provided exp param is either expired or is too much in the future.|
 |NodeIatOutsideGracePeriod|The provided iat param is either outside the grace_period or the issue time is in the future.|
 |NodeAuthFailed|The authMethod isn't allowed for the provided PKP.|
-|NodeInvalidAuthSig|The provided Authsig is invalid. Please ensure that the Authsig is in the required format as provided: https://js-sdk.litprotocol.com/interfaces/types_src.JsonAuthSig.html|
+|NodeInvalidAuthSig|The provided Authsig is invalid. Please ensure that the Authsig is in the required format as provided: https://js-sdk.litprotocol.com/interfaces/types_src.AuthSig.html|
```
Comment (debbly) [inline docs/SDK/Explanation/tests.md:56]: Same here with link styling, to be hyperlinked with `( here ) [ link ]`

File: docs/SDK/Explanation/tests.md [hunk 9]
```diff
@@ -53,10 +53,10 @@ Errors are thrown as exceptions when something has gone wrong. Errors are object
...
 |NodeAuthSigNotSupported|Multiple Authsigs not supported for the function.|
 |NodeInvalidEthereumSolanaAuthSig|Either the Authsig isn't provided or all the provided Authsigs are invalid.|
-|NodeInvalidED25519AuthSig|The provided Authsig isn't Ed25519 compliant. Please ensure that the Authsig is in the required format as provided: https://js-sdk.litprotocol.com/interfaces/types_src.JsonAuthSig.html|
+|NodeInvalidED25519AuthSig|The provided Authsig isn't Ed25519 compliant. Please ensure that the Authsig is in the required format as provided: https://js-sdk.litprotocol.com/interfaces/types_src.AuthSig.html|
```
Comment (debbly) [inline docs/SDK/Explanation/tests.md:59]: link styling

File: docs/SDK/Explanation/tests.md
Comment (debbly) [inline docs/SDK/Explanation/tests.md:1]: the link styling should be updated in this page, but given there are so many ... we can also do this in another PR

File: docs/pkp/minting.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+### Relying Parties and Supported Origins
+
+In order to allow for various frontends to integrate with our platform, we plan to support any domain to act as a [Relying Party](https://www.w3.org/TR/webauthn-2/#webauthn-relying-party) in the long run. However, we are in the process of slowly rolling out this authentication method currently maintain an allowlist of origins / domains that can integrate with the Lit network.
+
+### Challenge-Free Registration
+
```
Comment (debbly) [inline docs/pkp/minting.md:35]: What does a challenge mean? Can you add a definition.

File: docs/SDK/Explanation/tests.md
Comment (sarahzdev) [reply docs/SDK/Explanation/tests.md:1]: all set! @debbly

File: docs/SDK/Explanation/authentication/sessionSigs.md
Comment (sarahzdev) [reply docs/SDK/Explanation/authentication/sessionSigs.md:1]: done! mentioned this workflow in the readme.

File: docs/SDK/Explanation/authentication/methods.md (hunk 5 above)
Comment (hwrdtm) [reply docs/SDK/Explanation/authentication/methods.md]: I'm going to start a glossary section so there is one place for folks to look to for terms they're not familiar with.

File: docs/pkp/minting.md (hunk 3 above)
Comment (hwrdtm) [reply docs/pkp/minting.md:32]: think we removed the origin allowlist criteria, and RP can be anything now.

File: docs/SDK/Explanation/authentication/sessionSigs/use-cases.md [hunk 11]
```diff
@@ -0,0 +1,21 @@
+# Use Cases
+
+Below are some use cases of how to use session signatures and capability objects.
+
+_Note that these are not yet currently possible as we need to implement proof verification logic for delegation chains._
```
Comment (sarahzdev) [inline docs/SDK/Explanation/authentication/sessionSigs/use-cases.md:5]: might be awk to highlight these use cases if they aren't yet possible but open to what other's think

**Review by sarahzdev — APPROVED** [review]

File: docs/SDK/Explanation/authentication/sessionSigs/authMethods/webAuthn.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+      ability: LitAbility.AccessControlConditionDecryption
+    ],
+  },
+});
+```
+
+## Technical Details
+
+While the registration step involves minting a PKP via the Lit Relay Server (or, optionally, you can send a transaction to the contract yourself), we have implemented an authentication scheme that involves sending signed challenges to the decentralized Lit Network instead.
+
```
Comment (joshLong145) [inline docs/SDK/Explanation/authentication/sessionSigs/authMethods/webAuthn.md:87]: it might be nice to list here what data is needed to send to the contract when registering webauthn credentials in a bulleted list format. We break it down a bit below, but its a little wordy. maybe being explicit with example data so people can better understand what is needed if not minting through the relayer.

File: docs/SDK/Explanation/authentication/sessionSigs/getSessionSigs.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
+---
+sidebar_position: 2
+---
+
+# Generating SessionSigs
+
+You can use any wallet or auth method to generate session signatures with the `getSessionSigs()` function from the Lit SDK. This function generates a session keypair and uses a callback function that signs the generated session key to create an `AuthSig` that is scoped to specific capabilities.
+
```
Comment (joshLong145) [inline docs/SDK/Explanation/authentication/sessionSigs/getSessionSigs.md:8]: Since below we generate a wallet and hand role the `SIWE`  we should say they can use the `checkAndSignAuthMessage` function in the sdk if they are in a browser context.

**Review by joshLong145 — APPROVED** [review]
approving with some small comments. Looks Good!

File: docs/SDK/Explanation/authentication/sessionSigs/authMethods/socialLogin.md [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+// Begin login flow with Google
+async function authWithGoogle() {
+  const provider = litAuthClient.getProvider(
+    ProviderType.Google
+  );
+  await provider.signIn();
+}
+```
+
+Note: For Discord OAuth, you will initialize the provider with `ProviderType.Discord`.
```
Comment (debbly) [inline docs/SDK/Explanation/authentication/sessionSigs/authMethods/socialLogin.md:37]:
```suggestion
:::note
For Discord OAuth, you will initialize the provider with `ProviderType.Discord`.
:::
```

File: docs/SDK/Explanation/authentication/sessionSigs/capability-objects.md [hunk 15]
```diff
@@ -0,0 +1,58 @@
+---
+sidebar_position: 1
+---
```
Comment (debbly) [inline docs/SDK/Explanation/authentication/sessionSigs/capability-objects.md:3]: can remove the sidebar_position, since we are setting it in the `sidebarjs.js` file

File: docs/SDK/Explanation/authentication/sessionSigs/capability-objects.md
Comment (debbly) [inline docs/SDK/Explanation/authentication/sessionSigs/capability-objects.md:1]: file name is in snake case and so far the docs have been written in camel case - to follow the camel case logic I'd write the file as `capabilityObj`

File: docs/SDK/Explanation/authentication/sessionSigs/resources-and-abilities.md
Comment (debbly) [inline docs/SDK/Explanation/authentication/sessionSigs/resources-and-abilities.md:1]: file name is in snake case and so far the docs have been named in camel case - to follow the camel case logic I'd name the file as `resources&Abilities`

File: docs/SDK/Explanation/authentication/sessionSigs/use-cases.md
Comment (debbly) [inline docs/SDK/Explanation/authentication/sessionSigs/use-cases.md:1]: to follow the camel case logic I'd name the file as `useCases`

File: docs/SDK/Explanation/authentication/sessionSigs/use-cases.md (hunk 11 above)
Comment (debbly) [reply docs/SDK/Explanation/authentication/sessionSigs/use-cases.md:5]: I think this fits more within a post as a possibility with capabilities and session sigs!

I'm happy with keeping this page in the drafts until it's possible to do.

File: docs/pkp/minting.md [hunk 16]
```diff
@@ -0,0 +1,40 @@
+# Minting A PKP
+
+## Mint via Contracts
+
+You can mint a PKP NFT from our PKP contract on Polygon Mumbai [here](https://explorer.litprotocol.com/mint-pkp). This NFT represents the root ownership of the PKP. The NFT owner can grant other users (via a wallet address) or grant Lit Actions the ability to use the PKP to sign and decrypt data. They also have the ability to assign additional authentication methods, described at the bottom of the page.
```
Comment (debbly) [inline docs/pkp/minting.md:5]: Updating this to chronicle:

```suggestion
You can mint an NFT from our PKP contract on Chronicle, Lit's custom EVM rollup testnet, [here](https://explorer.litprotocol.com/mint-pkp). This NFT represents the root ownership of the PKP. The NFT owner can grant other users (via a wallet address) or grant Lit Actions the ability to use the PKP to sign and decrypt data. They also have the ability to assign additional authentication methods, described at the bottom of the page.

You can also use our handy auth helper contract on Chronicle [here](https://github.com/LIT-Protocol/LitNodeContracts/blob/main/contracts/PKPHelper.sol) and you can find the contract addresses [here](https://explorer.litprotocol.com/contracts).
```

File: docs/pkp/minting.md [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+We have a frontend that helps with this process at https://pkp-walletconnect.vercel.app/.
+
+We currently support both username-based and username-less WebAuthn registration, and usernames are purely used for your convenience / reference on the client-side.
+
+### Technical Details
+
+#### Contract Specifics
+
+- The `authMethodId` is derived from the credential's [rawId](https://www.w3.org/TR/webauthn-2/#dom-publickeycredential-rawid) parameter.
+- The `authMethodPubkey` is the COSE credential public key. **We currently only support Elliptic Curve COSE Key Type IDs**.
```
Comment (debbly) [inline docs/pkp/minting.md:28]:
```suggestion
- The `authMethodPubkey` is the [COSE credential public key](https://datatracker.ietf.org/doc/html/rfc8812). **We currently only support Elliptic Curve COSE Key Type IDs**.
```

link out to COSE and additional reading about it

File: docs/pkp/minting.md [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+
+#### Relying Parties and Supported Origins
+
+In order to allow for various frontends to integrate with our platform, we plan to support any domain to act as a [Relying Party](https://www.w3.org/TR/webauthn-2/#webauthn-relying-party) in the long run. However, we are in the process of slowly rolling out this authentication method currently maintain an allowlist of origins / domains that can integrate with the Lit network.
+
+#### Challenge-Free Registration
+
+We do not currently use challenges as part of our PKP minting / WebAuthn registration process and only use it for the PKP / WebAuthn authentication step.
```
Comment (debbly) [inline docs/pkp/minting.md:36]: let's also link to the glossary for challenges, since this is the first time mentioned if someone comes to this page 
```suggestion
We do not currently use [challenges](/resources/glossary#challenge) as part of our PKP minting / WebAuthn registration process and only use it for the PKP / WebAuthn authentication step.
```

File: docs/SDK/Explanation/authentication/sessionSigs/use-cases.md (hunk 11 above)
Comment (hwrdtm) [reply docs/SDK/Explanation/authentication/sessionSigs/use-cases.md:5]: okay, just going to remove it in sidebars

File: docs/SDK/Explanation/authentication/sessionSigs/resources-and-abilities.md
Comment (hwrdtm) [reply docs/SDK/Explanation/authentication/sessionSigs/resources-and-abilities.md:1]: I actually don't think we should go for camelCase, in part because it doesn't look good but here are some more substantial reasons: https://stackoverflow.com/questions/10302179/hyphen-underscore-or-camelcase-as-word-delimiter-in-uris

File: docs/SDK/Explanation/authentication/sessionSigs/use-cases.md
Comment (hwrdtm) [reply docs/SDK/Explanation/authentication/sessionSigs/use-cases.md:1]: see other comment

File: docs/SDK/Explanation/authentication/sessionSigs/capability-objects.md
Comment (hwrdtm) [reply docs/SDK/Explanation/authentication/sessionSigs/capability-objects.md:1]: see other comment

File: docs/SDK/Explanation/authentication/sessionSigs/authMethods/webAuthn.md (hunk 12 above)
Comment (hwrdtm) [reply docs/SDK/Explanation/authentication/sessionSigs/authMethods/webAuthn.md:87]: Will do but also, it's a bit hard to understand anyways since our contracts are still closed source and, unless their look on caldera + we publish & verify on there, it's pretty hard to get in plaintext.

**Review by sarahzdev — APPROVED** [review]
